app/controllers/users/phone_setup_controller.rb

```diff
@@ -135,5 +137,39 @@ def new_phone_form_params
         :recaptcha_mock_score,
       )
     end
+
+    def check_phone_submission_limit
+      if exceeded_phone_submission_limit?
+        return handle_too_many_phone_submissions
+      end
+      phone_submission_rate_limiter.increment
+      if exceeded_phone_submission_limit?
+        return handle_too_many_phone_submissions
+      end
+    end
+
+    def exceeded_phone_submission_limit?
+      if phone_submission_rate_limiter.exceeded_otp_send_limit?
+        return phone_submission_rate_limiter.lock_out_user
```

Comment:
I'm not sure I'd expect a side-effect in a method ending in `?`, which I'd normally expect to be pure and return a boolean value. That boolean value could be used elsewhere to do the lockout though? Though at that point I'm not sure the method has much value in place of just calling the methods on the rate-limiter directly inline (maybe even using [`delegate`](https://api.rubyonrails.org/classes/Module.html#method-i-delegate) to create short-hands).

Reply:
Removed this by reverting course.